soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -210,9 +222,9 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 		z_xt_ints_off(0xffffffff);
 		core_desc[cpu].bctl = DSPCS.bootctl[cpu].bctl;
 		DSPCS.bootctl[cpu].bctl &= ~DSPBR_BCTL_WAITIPCG;
-		soc_cpus_active[cpu] = false;
-		z_xtensa_cache_flush_inv_all();
 		if (cpu == 0) {
+			soc_cpus_active[cpu] = false;
+			z_xtensa_cache_flush_inv_all();
```

Comment:
couldn't this be done inside `save_core_context` too ? Because you are doing this big cache invalidation and then do other times in this function for smaller memory chunks. If you put it in the end you could do this flush/invalidation only once.

Reply:
In case of a secondary core I want to change this value only when everything is ready to turn it off.